crux_kv/src/lib.rs

```diff
@@ -21,6 +21,13 @@ pub enum KeyValueOperation {
     Delete { key: String },
     /// Test if a key exists
     Exists { key: String },
+    // List keys that start with a prefix, starting at the cursor
+    ListKeys {
+        /// The prefix to list keys for, or an empty string to list all keys
+        prefix: String,
+        /// The cursor to start listing from, or 0 to start from the beginning
+        cursor: u64,
```

Comment:
Not quite sure I thought it through enough, but would it make sense to be prefix OR cursor, but not both...? E.g. what happens if I reuse a cursor given to me for one prefix with a different prefix?

Presumably the shells would already need to keep the relevant cursor state around and that can include the original prefix, and that way the app doesn't need to worry about keeping the prefix around...?

Not sure... not quite clear on my own thinking, but red flags are going off in my head about the semantics now that I see it written down.

Reply:
I've introduced `KeyValueError::CursorNotFound`, and updated the doc comments to describe the semantics around how cursors should work from the shell's perspective